trio/_core/_multierror.py

```diff
@@ -174,21 +176,22 @@ class MultiError(BaseException):
 
     """
 
-    def __init__(self, exceptions):
+    def __init__(self, exceptions, *, _collapse=True):
         # Avoid recursion when exceptions[0] returned by __new__() happens
         # to be a MultiError and subsequently __init__() is called.
-        if hasattr(self, "exceptions"):
+        if _collapse and hasattr(self, "_exceptions"):
```

Comment:
Using the underscore-prefixed `_exceptions` name in this test feels like a backport-ism; does 3.11 have that name? I think you can probably use `hasattr(self, "exceptions")` since presumably the property will raise AttributeError if the underlying attribute doesn't exist.

Reply:
This comment is still relevant. I double-checked; here's 3.11:
```
>>> eg = ExceptionGroup("hi", [Exception("one"), Exception("two")])
>>> hasattr(eg, "exceptions")
True
>>> hasattr(eg, "_exceptions")
False
>>>
```